From 00c94530c0d57808ac1f9e9b1525341654278ba3 Mon Sep 17 00:00:00 2001
From: Mike Gorse <mgorse@suse.com>
Date: Fri, 7 Dec 2012 10:27:58 -0600
Subject: [PATCH 2/3] Fix another crash when removing hung processes

There was another instance where we were freeing the wrong data.
---
 registryd/deviceeventcontroller.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/registryd/deviceeventcontroller.c b/registryd/deviceeventcontroller.c
index 101dc6f..b14cd1e 100644
--- a/registryd/deviceeventcontroller.c
+++ b/registryd/deviceeventcontroller.c
@@ -1179,7 +1179,7 @@ reset_hung_process (DBusPendingCall *pending, void *data)
     if (!strcmp (l->data, dest))
     {
       g_free (l->data);
-      hung_processes = g_slist_remove (hung_processes, data);
+      hung_processes = g_slist_remove (hung_processes, l->data);
       break;
     }
   }
-- 
1.8.0

